graylog2-server/src/main/java/org/graylog2/plugin/Message.java

```diff
@@ -309,7 +317,7 @@ public void addField(final String key, final Object value) {
         } else if(value instanceof String) {
             final String str = ((String) value).trim();
 
-            if(!str.isEmpty()) {
+            if (isRequiredField || !str.isEmpty()) {
```

Comment:
I think we should still trim them to make the behavior consistent, we just should not let them be null (to prevent another breaking change and forcing all upstream producers to check for this, we could use `Strings.nullToEmpty` instead of rejecting the message construction by throwing an exception).